src/sonic-config-engine/tests/test_j2files.py

```diff
@@ -59,23 +69,24 @@ def create_machine_conf(self, platform, vendor):
                 'dell': 'onie',
                 'mellanox': 'onie'
                }
-        echo_cmd = "echo '{}_platform={}' | sudo tee -a /host/machine.conf > /dev/null".format(mode[vendor], platform)
+        echo_cmd1 = ["echo", '{}_platform={}'.format(mode[vendor], platform)]
+        echo_cmd2 = ["sudo", "tee", "-a", "/host/machine.conf"]
         if not os.path.exists('/host/machine.conf'):
             file_exist = False
             if not os.path.isdir('/host'):
                 dir_exist = False
-                os.system('sudo mkdir /host')
-            os.system('sudo touch /host/machine.conf')
-            os.system(echo_cmd)
+                subprocess.call(['sudo', 'mkdir', '/host'])
+            subprocess.call(['sudo', 'touch', '/host/machine.conf'])
+            getstatusoutput_noshell_pipe(echo_cmd1, echo_cmd2)
 
         return file_exist, dir_exist
 
     def remove_machine_conf(self, file_exist, dir_exist):
         if not file_exist:
-            os.system('sudo rm -f /host/machine.conf')
+            subprocess.call(['sudo', 'rm', '-f', '/host/machine.conf'])
```

Comment:
there's no user input in 'sudo rm -f /host/machine.conf', why is it a risk?

Reply:
It's better for extra security when using subprocess and with array of strings. https://semgrep.dev/docs/cheat-sheets/python-command-injection/#1c-using-os-module-to-execute-commands